src/commands/doctor.ts

```diff
@@ -227,22 +227,35 @@ ${this.doctor
     this.doctor.addCommandName(cmdString);
 
     const execPromise = new Promise<void>((resolve) => {
-      const execOptions = {
-        env: Object.assign({}, process.env),
-      };
-
-      exec(cmdString, execOptions, (error, stdout, stderr) => {
-        const code = error?.code || 0;
-        const stdoutWithCode = `Command exit code: ${code}\n\n${stdout}`;
-        const stdoutLogLocation = this.doctor.writeFileSync(
-          path.join(this.outputDir, 'command-stdout.log'),
-          stdoutWithCode
-        );
-        const debugLogLocation = this.doctor.writeFileSync(path.join(this.outputDir, 'command-debug.log'), stderr);
-        this.filesWrittenMsgs.push(`Wrote command stdout log to: ${stdoutLogLocation}`);
-        this.filesWrittenMsgs.push(`Wrote command debug log to: ${debugLogLocation}`);
-        resolve();
+      const stdoutLogLocation = this.doctor.getDoctoredFilePath(path.join(this.outputDir, 'command-stdout.log'));
+      const debugLogLocation = this.doctor.getDoctoredFilePath(path.join(this.outputDir, 'command-debug.log'));
+      this.doctor.createStdoutWriteStream(stdoutLogLocation);
+      this.doctor.createStderrWriteStream(debugLogLocation);
+      const cp = spawn(cmdString, [], { shell: true, env: Object.assign({}, process.env) });
+
+      this.filesWrittenMsgs.push(`Wrote command stdout log to: ${stdoutLogLocation}`);
```

Comment:
@shetzel putting these here is a hack. I had down in the on exit handler, but not sure how to make this wait in the promise all for the collected tasks so that the command execution is complete. Ideas?